### PR TITLE
Add feature to allow instructors to block students from viewing finalized submissions

### PR DESCRIPTION
**Pull request by lyskevin**
Screenshot of instructor's view of the toggle:
Screenshot 2021-08-23 at 11 14 36 AM

Screenshot of the student's view of a blocked submission:
Screenshot 2021-08-23 at 11 13 47 AM

### CODE REVIEW

File: client/app/bundles/course/assessment/containers/AssessmentForm/translations.intl.js [hunk 1]
```diff
@@ -33,6 +33,10 @@ const translations = defineMessages({
     id: 'course.assessment.form.autograded',
     defaultMessage: 'Autograded',
   },
+  blockStudentViewingAfterSubmitted: {
+    id: 'course.assessment.form.blockStudentViewingAfterSubmitted',
+    defaultMessage: 'Block Students from Viewing Finalized Submissions'
```
Comment (ekowidianto) [inline client/app/bundles/course/assessment/containers/AssessmentForm/translations.intl.js:38]: Missed a comma here

Comment (lyskevin) [reply client/app/bundles/course/assessment/containers/AssessmentForm/translations.intl.js]: Updated. Thanks!

File: .gitignore [hunk 2]
```diff
@@ -58,3 +58,6 @@ node_modules
 # Ignore build for client
 /client/build
 .DS_Store
+
+#Ignore local node version
```
Comment (zhuhanming) [inline .gitignore:62]: Add a space behind this `#`.

Also, should this be PRed separately? Should be a relatively safe PR that we can merge early and will improve dev experience.

File: config/locales/en/course/assessment/submission/submissions.yml [hunk 3]
```diff
@@ -9,6 +9,8 @@ en:
             failure: 'Could not create submission: %{error}'
           edit:
             attempt: :'course.assessment.assessments.assessment.attempt'
+          blocked:
+            message: 'This submission has been locked from viewing by the course staff.'
```
Comment (zhuhanming) [inline config/locales/en/course/assessment/submission/submissions.yml:13]: I feel like the message here can be improved. Maybe something like:

```
Submissions for this assessment cannot be viewed once finalised.
```

The current message might be misinterpreted as this specific student's submission being locked by course staff. Not sure if I'm overthinking this - I'm open to suggestions.

**Review by zhuhanming — COMMENTED** [review]
Generally LGTM! Do you mind adding a screenshot of the blocked/locked page to the PR as well?

File: app/views/course/assessment/assessments/_edit.json.jbuilder [hunk 4]
```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 json.attributes do
   json.(@assessment, :id, :title, :description, :start_at, :end_at, :bonus_end_at, :base_exp,
-    :time_bonus_exp, :published, :autograded, :show_private, :show_evaluation, :skippable,
-    :tabbed_view, :view_password, :session_password, :delayed_grade_publication, :tab_id,
-    :use_public, :use_private, :use_evaluation, :allow_partial_submission, :has_personal_times,
-    :affects_personal_times, :show_mcq_answer)
+        :time_bonus_exp, :published, :autograded, :show_private, :show_evaluation, :skippable,
```
Comment (zhuhanming) [inline app/views/course/assessment/assessments/_edit.json.jbuilder:4]: Sorry one last comment - what happened to the spacing for this part?

Comment (lyskevin) [reply app/views/course/assessment/assessments/_edit.json.jbuilder:4]: houndci-bot was complaining about the alignment, see here:
https://github.com/Coursemology/coursemology2/pull/4027#discussion_r692689310

File: .gitignore (hunk 2 above)
Comment (lyskevin) [reply .gitignore:62]: Good suggestion! Created a new PR here:
https://github.com/Coursemology/coursemology2/pull/4030

File: config/locales/en/course/assessment/submission/submissions.yml (hunk 3 above)
Comment (lyskevin) [reply config/locales/en/course/assessment/submission/submissions.yml]: Updated

**Comment by lyskevin** [comment]
Added screenshots of the instructor's view of the toggle and the student's view of a blocked submission

**Comment by zhuhanming** [comment]
Thanks for following up on my comments. Looks good to me, merging this PR.

**Pull request merged**